lib/valid_email2.rb

```diff
@@ -8,22 +8,35 @@ module ValidEmail2
   DISPOSABLE_FILE = File.expand_path('../config/disposable_email_domains.txt', __dir__)
 
   def self.disposable_emails
-    @disposable_emails ||= File.open(DISPOSABLE_FILE){ |f| f.read }.split("\n")
+    @disposable_emails ||= load_file(DISPOSABLE_FILE)
   end
 
   def self.blacklist
-    @blacklist ||= if File.exist?(BLACKLIST_FILE)
-                     YAML.load_file(File.expand_path(BLACKLIST_FILE))
-                   else
-                     []
-                   end
+    @blacklist ||= load_if_exists(BLACKLIST_FILE)
   end
 
   def self.whitelist
-    @whitelist ||= if File.exist?(WHITELIST_FILE)
-                     YAML.load_file(File.expand_path(WHITELIST_FILE))
-                   else
-                     []
-                   end
+    @whitelist ||= load_if_exists(WHITELIST_FILE)
+  end
+
+private
```

Comment:
`private` does not make class methods private in this context. It needs to be in a `class << self` or you can use `private_class_method`